catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -156,19 +158,63 @@ public void setMimeTypeMapper(MimeTypeMapper mimeTypeMapper) {
     }
 
     /**
-     * Retrieves a {@link ddf.catalog.resource.Resource} based on a {@link URI} and provided arguments. A connection is
-     * made to the {@link URI} to obtain the {@link ddf.catalog.resource.Resource}'s {@link InputStream} and build a
+     * Sets the directories that the {@link URLResourceReader} has permission to access when attempting to
+     * download a resource linked by a file URL.
+     * 
+     * @param rootResourceDirectoryPaths
+     *            a set of absolute paths specifying which directories the {@link URLResourceReader} has
+     *            permission to access when attempting to download resources linked by a file URL. A
+     *            null or empty input clears all root resource directory paths from the
+     *            {@link URLResourceReader} (this effectively blocks all resource downloads linked by file
+     *            URLs).
+     */
+    public void setRootResourceDirectories(Set<String> rootResourceDirectoryPaths) {
+        this.rootResourceDirectories.clear();
+        if (rootResourceDirectoryPaths != null) {
+            LOGGER.debug("Attempting to set Root Resource Directories to {} for {}",
+                    rootResourceDirectoryPaths.toString(), URLResourceReader.class.getSimpleName());
+
+            for (String rootResourceDirectoryPath : rootResourceDirectoryPaths) {
+                String path = null;
+                try {
+                    path = Paths.get(rootResourceDirectoryPath).toAbsolutePath().normalize()
+                            .toString();
+                    this.rootResourceDirectories.add(path);
+                    LOGGER.debug("Added [{}] to the list of Root Resource Directories for {}",
+                            path, URLResourceReader.class.getSimpleName());
+                } catch (InvalidPathException e) {
+                    LOGGER.error("{} is an invalid path.", rootResourceDirectoryPath, e);
+                }
+            }
+        }
```

Comment:
If rootResourceDirectoryPaths is null or blank, we could use KARAF_HOME as the default value.